pkg/kubelet/images/image_manager.go

```diff
@@ -125,6 +127,16 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 		Image:       image,
 		Annotations: podAnnotations,
 	}
+
+	// if RuntimeClassInImageCriAPI feature gate is enabled, set runtimeHandler CRI field
+	if utilfeature.DefaultFeatureGate.Enabled(features.RuntimeClassInImageCriAPI) {
+		runtimeHandler := ""
+		if pod.Spec.RuntimeClassName != nil && *pod.Spec.RuntimeClassName != "" {
+			runtimeHandler = *pod.Spec.RuntimeClassName
```

Comment:
could you check here.. might have to look it up: m.runtimeClassManager.LookupRuntimeHandler(pod.Spec.RuntimeClassName)

Reply:
@mikebrow I did figure out that we could go backwards and get the info. But my concern what rising from wanting to having uniformity with how images are being identified.

https://github.com/kubernetes/kubernetes/pull/121456#discussion_r1378092610

https://github.com/kubernetes/kubernetes/pull/121456#discussion_r1378094084

Reply:
unless the Spec.RuntimeClassName is the runtimeHandler.. I think you need to call LookupRuntimeHandler here

Reply:
change lgtm (passing the hander through Ensure)..